diff --git a/mantisbt/core/authentication_api.php b/mantisbt/core/authentication_api.php
index 720b7ed..fc2e975 100644
--- a/mantisbt/core/authentication_api.php
+++ b/mantisbt/core/authentication_api.php
@@ -760,8 +760,8 @@ function auth_get_password_max_size() {
 function auth_does_password_match( $p_user_id, $p_test_password ) {
 	$t_configured_login_method = config_get_global( 'login_method' );
 
-	if( LDAP == $t_configured_login_method ) {
-		return ldap_authenticate( $p_user_id, $p_test_password );
+	if (LDAP == $t_configured_login_method && ldap_authenticate($p_user_id, $p_test_password)) {
+		return TRUE;
 	}
 
 	if( !auth_can_use_standard_login( $p_user_id ) ) {
