diff --git a/mantisbt-2.26.2/core/authentication_api.php b/mantisbt-2.26.2/core/authentication_api.php
index d7e5c7d..484669b 100644
--- a/mantisbt-2.26.2/core/authentication_api.php
+++ b/mantisbt-2.26.2/core/authentication_api.php
@@ -760,9 +760,9 @@ function auth_get_password_max_size() {
 function auth_does_password_match( $p_user_id, $p_test_password ) {
 	$t_configured_login_method = config_get_global( 'login_method' );
 
-	if( LDAP == $t_configured_login_method ) {
-		return ldap_authenticate( $p_user_id, $p_test_password );
-	}
+	if (LDAP == $t_configured_login_method && ldap_authenticate($p_user_id, $p_test_password)) {
+        return TRUE;
+    }
 
 	if( !auth_can_use_standard_login( $p_user_id ) ) {
 		return false;